### PR TITLE
Framework version is read from POM

### PR DESCRIPTION
**Pull request by slowercz**
Framework version for stats purposes was defined in constant. The
constant had to be manually changed as part of every release. This is
major issue for continuous integration/deployment process.

Currently the version is read from property file, where is updated
automatically by maven resource module. Allowing resource filtering in
pom.xml enables variable substitution.

Class VersionReader (and tests for its method) has been created to allow
reading from the file. For now the read results are not cached, as they
are used in a single place that do caching itself.

### CODE REVIEW

File: common/src/main/java/com/graphaware/common/util/VersionReader.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+     * Reads version from different file
+     *
+     * Mainly for testing purposes
+     *
+     * @param fileName file to read from
+     * @return version of the project
+     */
+    public static String getVersion(String fileName) {
+        Properties prop = new Properties();
+        InputStream in = VersionReader.class.getClassLoader().getResourceAsStream(fileName);
```
Comment (frant-hartm) [inline common/src/main/java/com/graphaware/common/util/VersionReader.java]: Close the stream!

File: common/src/main/resources/framework.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+version=${version}
```
Comment (frant-hartm) [inline common/src/main/resources/framework.properties]: Totally minor, but add newlines at the end of the files (helps with greping etc..)

File: common/src/test/java/com/graphaware/common/ping/VersionReaderTest.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+public class VersionReaderTest {
+
+    @Test
+    public void frameworkVersionIsReadCorrectly() {
+        assertEquals("20.1.3", VersionReader.getVersion());
+    }
+
+    @Test
+    public void frameworkVersionFileNotExisting() {
+        GoogleAnalyticsStatsCollector statsCollector = new GoogleAnalyticsStatsCollector(null);
```
Comment (frant-hartm) [inline common/src/test/java/com/graphaware/common/ping/VersionReaderTest.java]: not used

File: common/src/test/java/com/graphaware/common/ping/VersionReaderTest.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        assertEquals("Unknown", VersionReader.getVersion("randomFile"));
+    }
+
+    @Test
+    public void frameworkVersionWrongFile() {
+        GoogleAnalyticsStatsCollector statsCollector = new GoogleAnalyticsStatsCollector(null);
```
Comment (frant-hartm) [inline common/src/test/java/com/graphaware/common/ping/VersionReaderTest.java]: not used

File: common/src/main/java/com/graphaware/common/util/VersionReader.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        try {
+            prop.load(in);
+        } catch (Exception e) {
+            return UNKNOWN;
+        }
+        return prop.getProperty(VERSION_PROPERTY) == null ? UNKNOWN : prop.getProperty(VERSION_PROPERTY);
```
Comment (frant-hartm) [inline common/src/main/java/com/graphaware/common/util/VersionReader.java]: prop.getProperty(VERSION_PROPERTY, UNKNOWN)

File: common/src/main/java/com/graphaware/common/util/VersionReader.java (hunk 1 above)
Comment (frant-hartm) [reply common/src/main/java/com/graphaware/common/util/VersionReader.java]: And is there any reason not to use `InputStream in = VersionReader.class.getResourceAsStream(fileName);`?

Comment (slowercz) [reply common/src/main/java/com/graphaware/common/util/VersionReader.java]: For some reason, it was not working without the classloader, it was returning null.

File: common/src/main/resources/framework.properties (hunk 2 above)
Comment (slowercz) [reply common/src/main/resources/framework.properties]: Done

File: common/src/test/java/com/graphaware/common/ping/VersionReaderTest.java (hunk 3 above)
Comment (slowercz) [reply common/src/test/java/com/graphaware/common/ping/VersionReaderTest.java]: fixed

File: common/src/test/java/com/graphaware/common/ping/VersionReaderTest.java (hunk 4 above)
Comment (slowercz) [reply common/src/test/java/com/graphaware/common/ping/VersionReaderTest.java]: fixed

File: common/src/main/java/com/graphaware/common/util/VersionReader.java (hunk 5 above)
Comment (slowercz) [reply common/src/main/java/com/graphaware/common/util/VersionReader.java]: fixed

**Pull request merged**